xwiki-platform-core/xwiki-platform-attachment/xwiki-platform-attachment-api/src/main/java/org/xwiki/attachment/internal/RedirectAttachmentClassDocumentInitializer.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.attachment.internal;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.reference.LocalDocumentReference;
+
+import com.xpn.xwiki.doc.AbstractMandatoryClassInitializer;
+import com.xpn.xwiki.objects.classes.BaseClass;
+
+import static java.util.Arrays.asList;
+
+/**
+ * Initialize the {@code Attachment.Code.RedirectAttachmentClass} XClass. This class is used to save the new location of
+ * moved attachments when a redirection must be preserved.
+ *
+ * @version $Id$
+ * @since 14.0RC1
+ */
+@Component
+@Named(RedirectAttachmentClassDocumentInitializer.HINT)
+@Singleton
+public class RedirectAttachmentClassDocumentInitializer extends AbstractMandatoryClassInitializer
+{
+    /**
+     * Hint for this class.
+     */
+    public static final String HINT = "Attachment.Code.RedirectAttachmentClass";
```

Comment:
This hint is public, why not using it in the named of the filter and of the listener?